modules/shared/src/main/scala/org/tessellation/schema/balance.scala

```diff
@@ -1,39 +1,63 @@
 package org.tessellation.schema
 
+import cats.kernel.Order
 import cats.syntax.either._
+import cats.syntax.semigroup._
 
 import scala.util.Try
 
 import derevo.cats.{eqv, show}
 import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
+import eu.timepit.refined.cats._
 import eu.timepit.refined.numeric.NonNegative
 import eu.timepit.refined.refineV
-import eu.timepit.refined.types.numeric.NonNegBigInt
+import eu.timepit.refined.types.numeric.NonNegLong
 import io.estatico.newtype.macros.newtype
-import io.estatico.newtype.ops._
 import io.getquill.MappedEncoding
 
 object balance {
 
   @derive(decoder, encoder, eqv, show)
   @newtype
-  case class Balance(value: NonNegBigInt)
+  case class Amount(value: NonNegLong)
+
+  @derive(decoder, encoder, eqv, show)
+  @newtype
+  case class Balance(value: NonNegLong) {
+
+    def plus(that: Amount): Either[BalanceOutOfRange, Balance] = {
+      val sum = value |+| that.value
+
+      if (Order[NonNegLong].gteqv(sum, value) && Order[NonNegLong].gteqv(sum, that.value)) {
+        Balance(sum).asRight[BalanceOutOfRange]
+      } else BalanceOutOfRange("Reached Long.MaxValue when adding balances!").asLeft[Balance]
+    }
+
+    def minus(that: Amount): Either[BalanceOutOfRange, Balance] =
+      NonNegLong.from(value.value - that.value.value).bimap(BalanceOutOfRange, Balance(_))
+  }
 
   object Balance {
-    val empty: Balance = Balance(NonNegBigInt.unsafeFrom(BigInt(0)))
+
+    val empty: Balance = Balance(NonNegLong(0L))
+
+    implicit def toAmount(balance: Balance): Amount = Amount(balance.value)
```

Comment:
It's not used and I guess not needed. We won't be adding/subtracting balances right?

Reply:
One use case would be to calculate total supply (we need to add all balances together). I'd keep that anyway